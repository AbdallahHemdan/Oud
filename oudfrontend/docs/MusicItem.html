<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: MusicItem</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: MusicItem</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>MusicItem<span class="signature">(props, item, id, name, icon)</span><span class="type-signature"></span></h2>
        
            <div class="class-description"><p>component to render all the Category stuff which is a list of playlist associated with this category</p></div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="MusicItem"><span class="type-signature"></span>new MusicItem<span class="signature">(props, item, id, name, icon)</span><span class="type-signature"></span></h4>
    

    











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>props</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last"><p>get props from higher components (Home)</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>item</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last"><p>get item from higher component (Home)</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>id</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last"><p>get id of an item from higher component (Home)</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>name</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last"><p>get name of an item from higher component (Home)</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>icon</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last"><p>get icon of an item from higher component (Home)</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    
    <dt class="tag-author">Author:</dt>
    <dd class="tag-author">
        <ul>
            <li>Abdallah Hemdan</li>
        </ul>
    </dd>
    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_MusicItem_MusicItem.js.html">components/MusicItem/MusicItem.js</a>, <a href="components_MusicItem_MusicItem.js.html#line26">line 26</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="componentDidMount"><span class="type-signature"></span>componentDidMount<span class="signature">()</span><span class="type-signature"> &rarr; {void}</span></h4>
    

    



<div class="description">
    <p>Fetching category playlists data immediately after the component has been mount to the DOM tree</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_MusicItem_MusicItem.js.html">components/MusicItem/MusicItem.js</a>, <a href="components_MusicItem_MusicItem.js.html#line72">line 72</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <ul>
<li>nothing to return, it just fetch data and set it in the state</li>
</ul>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">void</span>


    </dd>
</dl>

    





        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="About.html">About</a></li><li><a href="ActivityBar.html">ActivityBar</a></li><li><a href="Albums.html">Albums</a></li><li><a href="ArtistUserView.html">ArtistUserView</a></li><li><a href="GenreCard.html">GenreCard</a></li><li><a href="Home.html">Home</a></li><li><a href="LowerContainer.html">LowerContainer</a></li><li><a href="MainContent.html">MainContent</a></li><li><a href="MusicCard.html">MusicCard</a></li><li><a href="MusicItem.html">MusicItem</a></li><li><a href="Navbar.html">Navbar</a></li><li><a href="Options.html">Options</a></li><li><a href="Player.html">Player</a></li><li><a href="PlayingBarCenter.html">PlayingBarCenter</a></li><li><a href="PlayingBarLeft.html">PlayingBarLeft</a></li><li><a href="PlayingBarRight.html">PlayingBarRight</a></li><li><a href="Popular.html">Popular</a></li><li><a href="Queue.html">Queue</a></li><li><a href="RelatedArtists.html">RelatedArtists</a></li><li><a href="Sidebar.html">Sidebar</a></li><li><a href="SortableTrackContainer.html">SortableTrackContainer</a></li><li><a href="Track.html">Track</a></li><li><a href="TrackContainer.html">TrackContainer</a></li><li><a href="UpperContainer.html">UpperContainer</a></li><li><a href="WebPlayer.html">WebPlayer</a></li></ul><h3>Global</h3><ul><li><a href="global.html#addToQueue">addToQueue</a></li><li><a href="global.html#AfterLogin">AfterLogin</a></li><li><a href="global.html#BeforeLogin">BeforeLogin</a></li><li><a href="global.html#checkPassword">checkPassword</a></li><li><a href="global.html#deleteRequest">deleteRequest</a></li><li><a href="global.html#EditCountry">EditCountry</a></li><li><a href="global.html#EditGendr">EditGendr</a></li><li><a href="global.html#EditProfileTextElement">EditProfileTextElement</a></li><li><a href="global.html#FeedBack">FeedBack</a></li><li><a href="global.html#fetchCategoriesUrl">fetchCategoriesUrl</a></li><li><a href="global.html#Footer">Footer</a></li><li><a href="global.html#formValid">formValid</a></li><li><a href="global.html#HeaderBody">HeaderBody</a></li><li><a href="global.html#HeaderBodyBottom">HeaderBodyBottom</a></li><li><a href="global.html#HeaderBodyTop">HeaderBodyTop</a></li><li><a href="global.html#HelpUs">HelpUs</a></li><li><a href="global.html#HiddenSideBar">HiddenSideBar</a></li><li><a href="global.html#ImageContainer">ImageContainer</a></li><li><a href="global.html#MainContainer">MainContainer</a></li><li><a href="global.html#NavBar">NavBar</a></li><li><a href="global.html#PageContainer">PageContainer</a></li><li><a href="global.html#patchRequest">patchRequest</a></li><li><a href="global.html#pause">pause</a></li><li><a href="global.html#postRequest">postRequest</a></li><li><a href="global.html#Profile">Profile</a></li><li><a href="global.html#ProfileElement">ProfileElement</a></li><li><a href="global.html#putRequest">putRequest</a></li><li><a href="global.html#render">render</a></li><li><a href="global.html#resume">resume</a></li><li><a href="global.html#SideBarElement">SideBarElement</a></li><li><a href="global.html#SignOutEverywhere">SignOutEverywhere</a></li><li><a href="global.html#ThankYou">ThankYou</a></li><li><a href="global.html#YourPlan">YourPlan</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.3</a> on Mon Apr 20 2020 22:26:13 GMT+0200 (Eastern European Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>