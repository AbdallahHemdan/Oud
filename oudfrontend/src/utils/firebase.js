import firebase from "firebase";

const firebasConfig = {
  apiKey: "AIzaSyBZVUambAFEB-ywzAZTAPP-tSkbYXSEWc4",
  authDomain: "oud-notifications.firebaseapp.com",
  databaseURL: "https://oud-notifications.firebaseio.com",
  projectId: "oud-notifications",
  storageBucket: "oud-notifications.appspot.com",
  messagingSenderId: "681943816425",
  appId: "1:681943816425:web:9149a8834b2ff0f4623878"
};
// Initialize Firebase
firebase.initializeApp(firebasConfig);

export default firebase;
