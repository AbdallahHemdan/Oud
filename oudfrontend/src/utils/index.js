import axios from 'axios'
import { base } from '../config/environment'
const findByTestAttr = (component, val) => {
    return component.find(`[data-testid="${val}"]`);
}

function addToLikedSongs(id) {
    axios.put(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);

        })
        .catch(function (error) {
            console.log(error);
        });
}
function removeLikedSong(id) {
    axios.delete(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);
        })
        .catch(function (error) {
            console.log(error);
        });
}
export { findByTestAttr, addToLikedSongs, removeLikedSong }
