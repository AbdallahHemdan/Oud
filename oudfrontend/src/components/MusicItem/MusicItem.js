import React, { Component } from "react";
import "./MusicItem.css";
import { base } from "./../../config/environment";
import CategoryHeader from "./../CategoryHeader/CategoryHeader";
import CategoryBody from "./../CategoryBody/CategoryBody";
import axios from "axios";

/**
 * component to render all the Category stuff which is a list of playlist associated with this category
 *
 * @author Abdallah Hemdan
 *
 * @component
 *
 */
class MusicItem extends Component {
  /**
   * @constructor
   *
   * @param {object} props - get props from higher components (Home)
   * @param {object} item - get item from higher component (Home)
   * @param {object} _id - get _id of an item from higher component (Home)
   * @param {object} name - get name of an item from higher component (Home)
   * @param {object} icon - get icon of an item from higher component (Home)
  */
  constructor(props) {
    super(props);

    // destructuring item props
    const { _id, name, icon } = this.props.item;

    this.state = {
      /**
       * _id of the category
       *
       * @type {string}
       */

      _id: _id,

      /**
       * name of the category
       *
       * @type {string}
       */
      name: name,

      /**
       * icon of the category
       *
       * @type {string}
       */
      icon: icon,
      /**
       * List of playlists of category
       *
       * @type {Array<object>}
       */
      playlists: [],
    };
  }

  handleStoringPlaylists = ({ items, limit, offset, total }) => {
    this.setState({ playlists: items, limit, offset, total });
  };

  /**
   * Fetching category playlists data immediately after the component has been mount to the DOM tree
   *
   * @returns {void} - nothing to return, it just fetch data and set it in the state
  */
  componentDidMount() {
    let fetchPlaylistsUrlMocking = `${base}/browse/categories/${this.state._id}/playlists`;
    axios
      .get(fetchPlaylistsUrlMocking)
      .then((result) => {
        this.handleStoringPlaylists(result.data);
      })
      .catch((err) => {
        console.log(err);
      });
  }

  /**
   * @function
   * @name render
   * @description Render all the playlists of a specific category which it's defined by its _id
   *
   * @returns {JSX} Component for Home
   */
  render() {
    return (
      <div className="module" data-testid="music-item-container">
        <CategoryHeader
          name={this.state.name}
          _id={this.state._id}
          playlists={this.state.playlists}
          data-testid="category-header"
        />
        <CategoryBody
          webPlayer={this.props.webPlayer}
          playlists={this.state.playlists}
          data-testid="category-body"
        />
      </div>
    );
  }
}

export default MusicItem;
