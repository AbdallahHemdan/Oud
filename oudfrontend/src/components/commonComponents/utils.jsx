import axios from "axios";
import { base } from "../../config/environment"
import {config} from "../../utils/auth"
/**
 * @file this file contains three functions used in Playlist, Album, LikedSongs 
 **/
export function addSong(playlistId, track){
  axios
  .post(`${base}/playlists/${playlistId}/${track}`, config)
  .then((response) => {
  })
  .catch((error) => {
  });
}