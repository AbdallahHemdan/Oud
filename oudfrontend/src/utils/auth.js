import jwtDecode from "jwt-decode";
import { getRequest } from "./requester";
import { base } from "./../config/environment";

function getToken() {
  return localStorage.getItem("accessToken");
}

const config = {
  headers: { authorization: `Bearer ${getToken()}` }
};

function Auth() {
  try {
    if (!getToken()) return null;
    let decoded = jwtDecode(getToken());
    console.log(decoded.id);
    return decoded.id;
  } catch {
    return null;
  }
}
function isLoggedIn() {
  return getToken() ? true : false;
}

function isArtist() {
  return getRequest(`${base}/me`)
    .then(response => {
      return response.data.role === "artist";
    })
    .catch(error => {
      console.log(error);
    });
}
const _config = config;
export { _config as config };
export const token = getToken();
const _Auth = Auth;
export { _Auth as Auth };
const _isLoggedIn = isLoggedIn;
export { _isLoggedIn as isLoggedIn };
const _isArtist = isArtist;
export { _isArtist as isArtist };
