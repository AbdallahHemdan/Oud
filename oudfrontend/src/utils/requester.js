import axios from "axios";
import { isLoggedIn } from "./../utils/auth";
const config = {
  headers: {
    authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVlOTA3ZGIwYTA2NDVmNDU4MTYwNzYwNiIsImlhdCI6MTU5MTYyMTQxOSwiZXhwIjoxNTk0MjEzNDE5fQ.fj3N3Pc89Pf_xlt7fGmXw1SINTecUB4-y3pihAAPjC8`,
    "Access-Control-Allow-Origin": "*"
  }
};
const getRequest = endpoint => {
  return axios.get(endpoint, config);
};
/**
 * Axios DELETE request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @returns {object}
 */
const deleteRequest = endpoint => {
  return axios.delete(endpoint, config);
};
/**
 * Axios PUT request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const putRequest = (endpoint, body = {}) => {
  return axios.put(endpoint, body, config);
};
/**
 * Axios POST request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const postRequest = (endpoint, body = {}) => {
  return axios.post(endpoint, body, config);
};
/**
 * Axios PATCH request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const patchRequest = (endpoint, body = {}) => {
  return axios.patch(endpoint, body, config);
};

// const isArtist = () => {
//   if (isLoggedIn()) {
//   }
// };
export { getRequest, deleteRequest, putRequest, postRequest, patchRequest };
