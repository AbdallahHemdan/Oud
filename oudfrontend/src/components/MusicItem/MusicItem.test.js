import React from "react";
import { shallow } from "enzyme";
import MusicItem from "./MusicItem";
import { findByTestAttr } from "../../utils/index";

const setup = (props = {}) => {
  const component = shallow(<MusicItem {...props} />);
  return component;
};


const item = {
  item: {
    _id: 1,
    name: "Recently played",
    icon: "http://lorempixel.com/640/480/cats",
  },
};

describe("Music Item component", () => {
  let component;
  beforeEach(() => {
    const props = item;
    component = setup(props);
  });

  it('Should the category header in right way', () => {
    const wrapper = findByTestAttr(component, "category-header");
    expect(wrapper.length).toBe(1);
  });
  it('Should the category body in right way', () => {
    const wrapper = findByTestAttr(component, "category-body");
    expect(wrapper.length).toBe(1);
  });
  it('Should the music item container in right way', () => {
    const wrapper = findByTestAttr(component, "music-item-container");
    expect(wrapper.length).toBe(1);
  });

  describe('snapshot test', () => {
    it('renders component correctly', () => {
      const wrapper = shallow(
        <MusicItem item={item} />
      )
      expect(wrapper).toMatchSnapshot()
    });
  });

});
