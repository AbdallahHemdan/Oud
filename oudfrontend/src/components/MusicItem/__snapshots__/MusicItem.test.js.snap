// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Music Item component snapshot test renders component correctly 1`] = `ShallowWrapper {}`;
